booting_windows/readme.md

```diff
@@ -0,0 +1,76 @@
+# Running Windows over u-root
+
+## Prerequisites
+**WARNING**: the scripts will require sudo priviliges. It is highly advised that
+you read them carefully before running them!
+
+1. Golang, see `https://golang.org/`.
+1. Packages required to build Linux kernel
+1. A functional Windows Server 2019 image, asumed to exist at
+   `"${WORKSPACE}"/win2019.raw`. `setup.sh` will create a masking image over it,
+   so the original image will not be modified.
+1. An environment variable `EFI_WORKSPACE`, where files will be downloaded to or
+   otherwise created.
+
+## Installing the Modified u-root
+1.  Install u-root:
+
+    ```shell
+    go get github.com/u-root/u-root
+    ```
+
+1.  Change the uroot github remote to our modified one:
+
+    ```shell
+    pushd ~/go/src/github.com/u-root/u-root
+    git remote add oweisse https://github.com/oweisse/u-root  # our revised uroot repo
+    git fetch oweisse
+    git checkout -b kexec_test oweisse/kexec
+    go install
+    popd
+    ```
+
+## Setting up the Kernel.
+Setup Linux kernel source tree with our modifications. We modified
+kexec_load syscall to launch EFI applications. **Read the script before running!**
+
+The script will:
+1. Download an EFI loader image.
+1. Extract windows boot-manager from the windows image (see prerequisites above).
+1. Clone our forked linux kernel from `https://github.com/oweisse/linux/`
+   into $EFI_WORKSPACE.
+1. Install prerequisites (sudo required)
+1. Build Linux kernel
+
+```
+./setup.sh
+```
+
+## Running u-Root and booting Windows
+The script command line arguments:
+1. rebuild_uroot: will also add bootmgfw.efi, extracted from the Windows image
```

Comment:
I've noticed that it's essential to run rebuild_uroot on first invocation, whereas setup.sh built a kernel. For consistency and simplicity, perhaps remove the kernel build from setup.sh and just document that on first invocation it's necessary to pass both rebuild_kernel and rebuild_uroot?

Reply:
Done. See b28937d6